src/sql/workbench/parts/query/browser/actions.ts

```diff
@@ -205,12 +205,8 @@ export class ChartDataAction extends Action {
 	}
 
 	public run(context: IGridActionContext): Promise<boolean> {
-		let activeEditor = this.editorService.activeControl;
-		if (activeEditor instanceof QueryEditor) {
-			activeEditor.resultsEditor.chart({ batchId: context.batchId, resultId: context.resultId });
-			return Promise.resolve(true);
-		} else {
-			return Promise.resolve(false);
-		}
+		const activeEditor = this.editorService.activeControl as QueryEditor;
+		activeEditor.chart({ batchId: context.batchId, resultId: context.resultId });
+		return Promise.resolve(true);
```

Comment:
Should check undefined here in case the activeControl isn't a QueryEditor

Reply:
How we use it, it is. This action is a button in the query editor, so for this to be run query editor would have to be the active editor. I can't check for instance of Query Editor since it creates a circular dependency. Ideally these would all be contributions which would remove the circular dependency, but that would be a much larger work item.